niri-ipc/src/lib.rs

```diff
@@ -60,6 +62,8 @@ pub enum Response {
     FocusedWindow(Option<Window>),
     /// Output configuration change result.
     OutputConfigChanged(OutputConfigChanged),
+    /// Information about workspaces
+    Workspaces(Vec<IpcWorkspace>),
```

Comment:
I thought to maybe return an enum with a HashMap from output name to a vector of workspaces and a bare vector of workspaces as the second variant, but on a second thought this seems to work better (less chance to mess it up in the script with the rare "no outputs" case).

Reply:
I also started with a `HashMap` but thought this would be cleaner in the long run.